qiskit_experiments/characterization/t2ramsey.py

```diff
@@ -99,3 +102,203 @@ def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+    duration of the delay. By scanning this duration, we can get
+
+    .. math::
+
+        {\cal E}_x(\tau)
+            = {\rm Re} {\rm Tr}\left( \sigma_Y U \rho U^\dagger \right)
+            &= - \cos(\Delta\omega\tau) = \sin(\Delta\omega\tau - \frac{\pi}{2}), \\
+        {\cal E}_y(\tau)
+            = {\rm Re} {\rm Tr}\left( \sigma_X U \rho U^\dagger \right)
+            &= \sin(\Delta\omega\tau),
+
+    where :math:`\rho = | L \rangle` prepared by the :math:`\sqrt{\rm X}` gate in the beginning.
+
+    Note that phase difference of these two outcomes :math:`{\cal E}_x, {\cal E}_y` depends on
+    the frequency offset :math:`\Delta\omega`, which is also sensitive
+    to the sign of it, in contrast to the standard Ramsey experiment which
+    usually consists only of the first circuit,
+    i.e. :math:`\cos(-\Delta\omega\tau) = \cos(\Delta\omega\tau)`.
+    """
+
+    __analysis_class__ = RamseyXYAnalysis
+
+    def __init__(
+            self,
+            qubit: int,
+            delays: Union[Iterable[float], np.ndarray],
+            unit: str = "s",
+            offset_frequency: float = 0.,
+    ):
+        """Initialize new experiment.
+
+        Args:
+            qubit: The qubit under test.
+            delays: Delay times of the experiments.
+            unit: Time unit of `delays`. One of 's', 'ms', 'us', 'ns', 'ps', 'dt'.
+            offset_frequency: Frequency offset for this experiment.
+
+        .. warning::
+
+            ``offset_frequency`` doesn't really change the qubit frequency for now,
+            because of the missing oscillator frequency upgrade pass in the circuit execution.
+        """
+        super().__init__([qubit])
+
+        default_options = self.experiment_options
+        default_options.delays = delays
+        default_options.unit = unit
+        default_options.offset_frequency = offset_frequency
+
+        self.set_experiment_options(**default_options.__dict__)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        return Options(
+            delays=None,
+            unit="s",
+            offset_frequency=0.,
+            calibrations={
+                "sx": None,
+                "measure": None,
+            },
+        )
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Return a list of experiment circuits.
+
+        Args:
+            backend: Optional, a backend object.
+
+        Returns:
+            A list of :class:`QuantumCircuit`.
+        """
+        options = self.experiment_options
+
+        p_delay = Parameter("delay")
+
+        ram_x = QuantumCircuit(1)
+        ram_x.sx(0)
+        ram_x.delay(p_delay, 0, unit=options.unit)
+        ram_x.sx(0)
+        ram_x.measure_active()
+
+        ram_y = QuantumCircuit(1)
+        ram_y.sx(0)
+        ram_y.delay(p_delay, 0, unit=options.unit)
+        ram_y.rz(-np.pi/2, 0)
+        ram_y.sx(0)
+        ram_y.measure_active()
```

Comment:
What if someone wants to do RamseyXY on the qutrit 1 to 2 transition. There will be a lot of rewriting to do. Can you make this more flexible?

Reply:
Good suggestion. Do we add pre/post circuit which is defined by the method that can be overridden by subclasses?

Reply:
I think that is a good approach. So far that worked for EFRabi and Spec on 1-2